pages/data-user/perangkat-desa/detail/_id/index.vue

```diff
@@ -0,0 +1,26 @@
+<template>
+  <PerangkatDesaDetail :item="data" @updated="updated" />
+</template>
+
+<script>
+export default {
+  name: 'PagesPerangkatDesaDetail',
+  layout: 'Dashboard',
+  async asyncData ({ params, $axios }) {
+    try {
+      const response = await $axios.get(`/users/${params.id}`)
+      const data = response.data?.data
+      return {
+        data
+      }
+    } catch (error) {
+      error({ message: 'Gagal memuat data' })
```

Comment:
ask/discussion:

Hello, @doohanas is this code working as intended? As far as I know, Axios only return an Error Object to catch block if they fail to fetch data from API.

Maybe what you want is an `error` context from Nuxt.js to show their built-in error layout>. If that's the case, you need to add `error` as a parameter to `asyncData` like this:

```js
async asyncData ({ params, $axios, error }) {
  try {
    ...
  } catch (err) {
     error({ message: err.message }) // this code will tell Nuxt.js to show error layout with a message as a props
  }

```

CMIIW 🤔 

ref:
https://axios-http.com/docs/handling_errors
https://nuxtjs.org/docs/internals-glossary/context/#error

Reply:
Hello @Ibwedagama, Thank you for your insight. Yes, when I try it the message isn't showing. How if i write like this.

```js
async asyncData ({ params, $axios, error }) {
  try {
    ...
  } catch {
     error({ message: 'Gagal memuat data' })
  }
```